### PR TITLE
Fix logic so check will pass correctly

### PR DESCRIPTION
**Pull request by whasf**

### CODE REVIEW

**Comment by TeoTwawki** [comment]
had muh ternary results backwards

File: src/map/lua/lua_baseentity.cpp [hunk 1]
```diff
@@ -8734,8 +8734,8 @@ inline int32 CLuaBaseEntity::checkKillCredit(lua_State *L)
 
     bool credit = false;
     int lvlDiff = (int)(PMob->GetMLevel()) - (int)(PChar->GetMLevel());
-    int maxDiff = (lua_isnil(L, 2) || !lua_isnumber(L, 2)) ? (int)lua_tonumber(L, 2) : 15;
-    float range = (lua_isnil(L, 3) || !lua_isnumber(L, 3)) ? (float)lua_tonumber(L, 3) : 100;
+    int maxDiff = (!lua_isnil(L, 2) && lua_isnumber(L, 2)) ? (int)lua_tonumber(L, 2) : 15;
```
Comment (ibm2431) [inline src/map/lua/lua_baseentity.cpp]: Does `lua_isnumber` actually return true if the value is nil?

Comment (whasf) [reply src/map/lua/lua_baseentity.cpp]: Yes
**https://pgl.yoyo.org/luai/i/lua_isnumber**

Comment (ibm2431) [reply src/map/lua/lua_baseentity.cpp]: But.. but.. _nil_ isn't a string! :cry:

**Pull request merged**